src/analysis_and_optimization/Monotone_framework.ml

```diff
@@ -52,9 +53,11 @@ and free_vars_fnapp kind l =
 let rec free_vars_stmt (s : (Expr.Typed.t, Stmt.Located.t) Stmt.Fixed.Pattern.t)
     =
   match s with
-  | Assignment ((_, _, []), e) | Return (Some e) | TargetPE e ->
+  | Assignment ((LVariable _ | LTupleProjection _), _, e)
+   |Return (Some e)
+   |TargetPE e ->
       free_vars_expr e
-  | Assignment ((_, _, l), e) ->
+  | Assignment (LIndexed (_, l), _, e) ->
       Set.Poly.union_list (free_vars_expr e :: List.map ~f:free_vars_idx l)
```

Comment:
Yes I believe @nhuurre is right and we should recurse on the indexed expression here; `i` is a free variable in that example (and `i` would be found if the expression were `x.1[i][j]`, which would ideally behave the same way)

Reply:
Thanks! I updated this and the logic further down the file which finds usages.

Reply:
The optimization tests showed no difference with this change, so if you have any recommendations for things we could do which would exercise this it would be nice to include them